tests/project/test_project.py

```diff
@@ -126,6 +129,30 @@ def test_maintenance_get_repository_status(self):
         df = self.project.maintenance.get_repository_status()
         self.assertIn('pyiron_base', df.Module.values)
 
+    def test_load(self):
+        loaded = self.project.load("toy_1")
+        self.assertIsInstance(loaded, ToyJob, msg="Expected to load the full object")
+
+        self.assertEqual(
+            len(self.project.job_table(recursive=False)),
+            len(self.project.load.__dir__()),
+            msg="Tab completion (`__dir__`) should see both jobs at this project level"
+        )
+
+        not_fully_loaded = self.project.load("toy_1", convert_to_object=False)
+        self.assertIsInstance(
+            not_fully_loaded,
+            JobPath,
+            msg="Expected to load only an HDF interface"
+        )
+
+        inspected = self.project.inspect("toy_1")
+        self.assertIsInstance(
+            inspected,
+            JobPath,
+            msg="Expected to load only an HDF interface"
+        )
```

Comment:
Does it makes sense to test that `__getattr__` actually works, i.e. `self.assertEqual(self.project.inspect.toy_1, self.project.load('toy_1'))`?

Reply:
Yeah, good idea. I'm just on my phone but I'll do that next time I'm working.

Reply:
Turns out it doesn't work because neither the `JobPath` nor `GenericJob` are playing nicely with equality comparison. Instead I'm just testing that the result has the same `.name` and added a `TODO` to compare the objects directly sometime in the future once that's supported.